test/xdotoolify-spec.js

```diff
@@ -196,4 +196,74 @@ describe('xdotoolify', function() {
     
     expect(errorMsg).toBe('You forgot to add ".do() "at the end of a subcommand.');
   }));
+
+  it('should throw error when missing checkUntil after interaction', syncify(async function() {
+    let errorMsg = 'Nothing thrown';
+    let goodFunc = Xdotoolify.setupWithPage((page) => { return 5; });
+    let withCheck = Xdotoolify.setupWithPage((page) => {
+      return page.X
+          .click()
+          .checkUntil(goodFunc, x => x * 2, 10)
+          .do()
+    });
+
+    try {
+      await page.X
+          .run(withCheck)
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
+
+    expect(errorMsg).toBe('Nothing thrown');
+
+    withCheck = Xdotoolify.setupWithPage((page) => {
+      return page.X
+          .click()
+          .checkNothing()
+          .do()
+    });
+
+    try {
+      await page.X
+          .run(withCheck)
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
+
+    expect(errorMsg).toBe('Nothing thrown');
+
+    let withoutCheck = Xdotoolify.setupWithPage((page) => {
+      return page.X
+          .click()
+          .do({unsafe: true});
+    });
+
+    try {
+      await page.X
+          .run(withoutCheck)
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
+    
+    expect(errorMsg).toBe('Nothing thrown');
+
+    withoutCheck = Xdotoolify.setupWithPage((page) => {
+      return page.X
+          .click()
+          .do();
+    });
+
+    try {
+      await page.X
+          .run(withoutCheck)
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
+    
+    expect(errorMsg).toBe('Missing checkUntil after interaction.');
```

Comment:
What does the full error trace look like here? Does it point to exactly which line is at fault? Would be annoying to get this line with an inaccurate stack trace.

Reply:
I tried it out before and it does trace back to the latest xdotoolify statement before the failure (similar to how it works with timeouts).